`node:` prefixes cause a warning

Running `nodeFileTrace` on an app that imports Node built-ins with the `node:` prefix (which is the preferred way to import built-ins as of Node 18) causes a warning to be emitted:

```js
import fs from 'node:fs';
```

```
warnings: Set(1) {
  Error: Failed to resolve dependency node:fs:
  Cannot find module 'node:fs' loaded from [dir]/src/index.js
      at [dir]/node_modules/@vercel/nft/out/node-file-trace.js:304:39
      at async Promise.all (index 0)
      at async Job.emitDependency ([dir]/node_modules/@vercel/nft/out/node-file-trace.js:290:9)
      at async Promise.all (index 0)
      at async nodeFileTrace ([dir]/node_modules/@vercel/nft/out/node-file-trace.js:32:5)
      at async file://[dir]/index.js:3:13
```

Removing the `node:` prefix fixes it, but that's not always an option since it could be in library code.

Reproduction here: https://github.com/Rich-Harris/vercel-nft-repro

--- a/src/analyze.ts
+++ b/src/analyze.ts
@@ -302,7 +302,7 @@ export default async function analyze(id: string, code: string, job: Job): Promi
         value: {
           [FUNCTION] (specifier: string) {
             deps.add(specifier);
-            const m = staticModules[specifier];
+            const m = staticModules[specifier.startsWith('node:') ? specifier.slice(5) : specifier];
             return m.default;
           },
           resolve (specifier: string) {
@@ -342,7 +342,7 @@ export default async function analyze(id: string, code: string, job: Job): Promi
       if (decl.type === 'ImportDeclaration') {
         const source = String(decl.source.value);
         deps.add(source);
-        const staticModule = staticModules[source];
+        const staticModule = staticModules[source.startsWith('node:') ? source.slice(5) : source];
         if (staticModule) {
           for (const impt of decl.specifiers) {
             if (impt.type === 'ImportNamespaceSpecifier')

--- a/src/resolve-dependency.ts
+++ b/src/resolve-dependency.ts
@@ -139,7 +139,7 @@ function resolveExportsImports (pkgPath: string, obj: PackageTarget, subpath: st
   } else {
     matchObj = obj;
   }
-  
+
   if (subpath in matchObj) {
     const target = getExportsTarget(matchObj[subpath], job.conditions, cjsResolve);
     if (typeof target === 'string' && target.startsWith('./'))
@@ -189,9 +189,10 @@ async function packageImportsResolve (name: string, parent: string, job: Job, cj
 async function resolvePackage (name: string, parent: string, job: Job, cjsResolve: boolean): Promise<string | string []> {
   let packageParent = parent;
   if (nodeBuiltins.has(name)) return 'node:' + name;
+  if (name.startsWith('node:')) return name;
 
   const pkgName = getPkgName(name) || '';
-  
+
   // package own name resolution
   let selfResolved: string | undefined;
   if (job.conditions) {

--- a/test/unit/asset-fs-array-expr-node-require/asset1.txt
+++ b/test/unit/asset-fs-array-expr-node-require/asset1.txt
@@ -1,0 +1,1 @@
+hello world

--- a/test/unit/asset-fs-array-expr-node-require/input.js
+++ b/test/unit/asset-fs-array-expr-node-require/input.js
@@ -1,4 +1,4 @@
-const { spawn } = require('child_process');
-const { join } = require('path');
+const { spawn } = require('node:child_process');
+const { join } = require('node:path');
 
 const child = spawn('gifsicle', ['--colors', '256', join(__dirname, './asset1.txt')]);

--- a/test/unit/asset-fs-array-expr-node-require/output.js
+++ b/test/unit/asset-fs-array-expr-node-require/output.js
@@ -1,0 +1,5 @@
+[
+  "package.json",
+  "test/unit/asset-fs-array-expr-node-require/asset1.txt",
+  "test/unit/asset-fs-array-expr-node-require/input.js"
+]

--- a/test/unit/asset-fs-array-expr/input.js
+++ b/test/unit/asset-fs-array-expr/input.js
@@ -1,4 +1,4 @@
 const { spawn } = require('child_process');
-const { join } = require('path');
+const { join } = require('node:path');
 
 const child = spawn('gifsicle', ['--colors', '256', join(__dirname, './asset1.txt')]);

--- a/test/unit/asset-fs-inline-path-enc-es-node-import/asset.txt
+++ b/test/unit/asset-fs-inline-path-enc-es-node-import/asset.txt
@@ -1,0 +1,1 @@
+hello world

--- a/test/unit/asset-fs-inline-path-enc-es-node-import/input.js
+++ b/test/unit/asset-fs-inline-path-enc-es-node-import/input.js
@@ -1,4 +1,4 @@
-import fs from 'node:fs';
-import * as path from 'node:path';
+import fs from 'fs';
+import * as path from 'path';
 
 console.log(fs.readFileSync(path.join(__dirname, 'asset.txt'), 'utf8'));

--- a/test/unit/asset-fs-inline-path-enc-es-node-import/output.js
+++ b/test/unit/asset-fs-inline-path-enc-es-node-import/output.js
@@ -1,0 +1,5 @@
+[
+  "package.json",
+  "test/unit/asset-fs-inline-path-enc-es-node-import/asset.txt",
+  "test/unit/asset-fs-inline-path-enc-es-node-import/input.js"
+]

